### PR TITLE
Drop superfluous copies from vala

### PR DESCRIPTION
**Pull request by ricotz**

### CODE REVIEW

File: config.vala [hunk 1]
```diff
@@ -1 +1 @@
-const string version_string = "2.1-1.4.0-5-g700a887";
+const string version_string = "?-1.5.0";
```
Comment (radare) [inline config.vala]: that looks wrong

File: Makefile [hunk 2]
```diff
@@ -67,11 +64,7 @@ $(BIN).exe: $(SRC) | $(VAPIS)
 
 $(BIN): $(SRC) | $(VAPIS)
 	@echo 'Compiling $(VALA_FILTER) -> $@'
-ifeq ($(VALAPKG),libvala-0.40)
-	$(VALAC) --vapidir=/usr/local/share/vala/vapi -o $@ --pkg posix $(VALA040) --save-temps ${TEMPS}
-else
-	$(VALAC) -o $@ --pkg posix --pkg $(VALAPKG) --pkg valacodegen-0.40 --save-temps ${TEMPS}
-endif
+	$(VALAC) -o $@ --pkg posix $(VALA_PRIVATE_CODEGEN) --save-temps ${TEMPS}
```
Comment (radare) [inline Makefile]: that will break in old valas

File: config.vala (hunk 1 above)
Comment (ricotz) [reply config.vala]: Oh, will drop this change

File: Makefile (hunk 2 above)
Comment (ricotz) [reply Makefile]: Using vala < 0.40 was already broken before afaics
Obviously this change will work with vala 0.38.5+ only, so additional conditions are required

**Comment by radare** [comment]
Yeah, im aware of that brokeness.. we can just drop old vala version support. because this current code works on <0.35.

we can drop all that double-vala support because most ppl will just install last release from distro and find out no single version of valabind works for them.. and ppl can still download old valabind to build against.

**Comment by radare** [comment]
Doesnt builds because of this:

ld: warning: directory not found for option '-L-lvalaccodegen'

**Comment by radare** [comment]
1:~$ pkg-config --variable pkglibdir libvala-0.40

0:~$

**Comment by radare** [comment]
this line is wrong VALA_PRIVATE_CODEGEN+=-X -L$(shell pkg-config --variable=pkglibdir lib$(shell ./getvv)) -X -lvalaccodegen

**Comment by radare** [comment]
just libdir is enough, pkglibdir doesnt exist

**Comment by radare** [comment]
but well, libdir is not the correct var either. so basically the pkgconfig doesnt provides a way to get that path. which should be $LIBDIR/vala-0.40/libvalaccodegen.dylib

**Comment by radare** [comment]
i have this fixed in local, ive merged your pr and i'll push evertyhing now.

Thanks!

**Comment by ricotz** [comment]
Please try to read more carefully, as I said this works with valac 0.38.5+ and valac 0.39.5+
which has the private library and pkglibdir exists in pkg-config file.

**Comment by radare** [comment]
i know that vala 0.38.5+ exposes the private library, but vala <0.34 exposed the private api without the ccodegen. maybe i didnt explained well.

i have 0.39.2 installed and there's no pkglibdir.

**Comment by ricotz** [comment]
When I have pushed it I explicitly linked you to the commit which introduced pkglibdir variable to make this easier for you. Not sure how I can make it more clear which version is required.

**Comment by radare** [comment]
i've fixed this in master, it's now using the pkg-config method and the other way as fallback

**Pull request closed without merging**